lua/paint/highlight.lua

```diff
@@ -37,6 +39,9 @@ function M.highlight(buf, first, last)
           from - 1,
           { end_col = to, hl_group = hl.hl, priority = 110 }
         )
+
+        last_to = to
```

Comment:
best to make this `to+1`, just to be sure we don't enter an endless loop. Could probably only happen when highlighting a single character, but still :)

Reply:
Right, did that, but I still have an infinite loop somewhere (that's why I initially marked it as draft), I'm trying to debug it.